### DISCUSSION TITLE
Errors when Loading Alpine from UNPKG

### DISCUSSION THREAD

**Discussion by bobwurtz** [alpinejs/alpine#1558] [7. Bugs (archived)]
Hey All,

I recently needed to update to the latest version of Alpine, so I followed the recommended steps and removed it from `npm` and loaded it in the `<head>` of my pages, using:

`<script defer src="https://unpkg.com/alpinejs@3.0.6/dist/cdn.min.js"></script>`

After I made this switch, I started getting _**A LOT**_ of error messages in the console. When the page initially loads, I get

```
Uncaught RangeError: Maximum call stack size exceeded.
    at dispatch (dispatch.js:6)
    at Livewire.value (index.js:87)
    at sale?location_id=1:52
    at B (cdn.min.js:5)
    at Object.Kt [as start] (cdn.min.js:5)
    at app.js:1941
    at sale?location_id=1:44
    at dispatch (dispatch.js:6)
    at Livewire.value (index.js:87)
    at sale?location_id=1:52

dispatch.js:6
```

and I also get 21 of these: `cdn.min.js:5 Uncaught (in promise) TypeError: l is not a function at cdn.min.js:5`

When I trigger a Livewire event to reset some values on the page, I get dozens of these:

`cdn.min.js:5 Uncaught (in promise) {isFromCancelledTransition: true}`

and then another 21 of `cdn.min.js:5 Uncaught (in promise) TypeError: l is not a function at cdn.min.js:5`, followed by another lone instance of `cdn.min.js:5 Uncaught (in promise) TypeError: l is not a function at cdn.min.js:5`. None of my Livewire elements work and the page does not function.

When I switch back to using `npm`, I have no problems. Anyone else experience this?

Livewire: 2.5.1
Alpine: 3.0.6

**Comment by KevinBatdorf** [comment]
Can you use the non minified CDN version and see if you can get better feedback?

**Reply by calebporzio** [reply]
What Kevin said, but also, can you post a stripped down version to reproduce?

**Reply by bobwurtz** [reply]
I get the same error messages with the non-minified version but now they have more detail.

`Uncaught (in promise) TypeError: i is not a function at cdn.js:1675`

I attempted to reproduce this in a fresh Laravel install and was not able to do so. Loading from the CDN works fine in the fresh install. So, something about my app must be causing this, right? This error is new - started appearing last week.

Again, when I use npm locally everything is fine.

But now something else strange has started. I use Vapor to run my app and this morning I started getting this error in staging:

```
Uncaught (in promise) TypeError: Cannot read property 'version' of undefined
    at beforeAlpineTwoPointSevenPointThree (SupportAlpine.js:315)
    at isHiding (SupportAlpine.js:299)
    at alpinifyElementsForMorphdom (SupportAlpine.js:274)
    at onBeforeElUpdated (index.js:459)
    at callHook (morphdom.js:35)
    at morphEl (morphdom.js:199)
    at morphdom.js:332
    at morphEl (morphdom.js:219)
    at morphdom.js:463
    at Component.value (index.js:366)
```

It looks like my staging environment doesn't realize that it's on version 3 now. Running the same npm install of Alpine locally does not produce this error. I think this is a long-winded way of saying that I screwed up something about my Alpine install/settings at some point and I need to wipe them and start over.

How do I do that? I've tried uninstalling and reinstalling Alpine from npm but that hasn't worked yet.

Edit: I'm not sure if this matters, but the `'version' of undefined` error is being triggered by a Livewire.emit event. And I was wrong, it also affects my local machine. I feel like I need to reset something but I don't know what.

**Comment by bobwurtz** [comment]
Update: after uninstalling and reinstalling alpinejs using `npm` about a dozen times, I uninstalled and now it is loaded using the suggested method:

`<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>`

I had tried this before as well but without any luck.

I ran `npm update` at one point, which I read might help. I'm not sure what did it exactly, but it's working now. The console error messages have disappeared as well.

Alpine is an amazing project. Thanks for everything you do.
